src/typing/typer.ml

```diff
@@ -281,6 +281,17 @@ let rec type_ident_raise ctx i p mode with_type =
 			AKExpr (get_this ctx p)
 		| (MCall _, KAbstractImpl _) | (MGet, _)-> AKExpr(get_this ctx p)
 		| _ -> AKNo i)
+	| "abstract" ->
+		begin match mode, ctx.curclass.cl_kind with
+			| MSet _, KAbstractImpl ab -> typing_error "Property 'abstract' is read-only" p;
+			| (MGet, KAbstractImpl ab)
+			| (MCall _, KAbstractImpl ab) ->
+				let t = TAbstract (ab,[]) in
+				let block = mk (TBlock [(get_this ctx p)]) t p in
```

Comment:
Are you sure `TBlock` is required here?

Reply:
`mk (TCast ((get_this ctx p), None)) t p` also works, but ide hint for `abstract` will be `this:String` instead of `MyAbstract`. Is there a better way to type `(get_this ctx p)` with `TAbstract`?

Reply:
Try this:
https://github.com/HaxeFoundation/haxe/blob/3fac24b5e2465af7ae26fb533f5ef6d93cf29825/src/typing/typer.ml#L340-L341

Reply:
Cool, but still looks like `this:String` or `this:T` instead of `MyAbstract` on hover. Maybe this can be improved in display stuff someway, but maybe `TBlock` is fine?

Reply:
Have you tried with `cast` instead of `TBlock`?

Reply:
> Maybe this can be improved in display stuff someway
Hmm... I don't know. Maybe @Simn knows. But `TBlock` is out of place here.

Reply:
Yes let's not add random `TBlock` nodes like that. I can check the display situation, maybe you could add a failing test for it.